src/pyfmi/fmi.pyx

```diff
@@ -8222,12 +8261,49 @@ cdef class FMUModel(FMUModelME1):
     def __init__(self, fmu, path='.', enable_logging=True):
         print("WARNING: This class is deprecated and has been superseded with FMUModelME1. The recommended entry-point for loading an FMU is now the function load_fmu.")
         FMUModelME1.__init__(self,fmu,path,enable_logging)
-        
+
 def _handle_load_fmu_exception(fmu, log_data):
     for log in log_data:
         print(log)
 
-def load_fmu(fmu, path = '.', enable_logging = None, log_file_name = "", kind = 'auto', log_level=FMI_DEFAULT_LOG_LEVEL):
+def display_path_deprecation(path):
+    if path != '.':
+        w = "The argument 'path' is deprecated and will be removed in a future version."
+        w += " Please specify a full path to fmu via the argument 'fmu' instead."
+        logging.warning(w)
+
+def check_fmu_args(allow_unzipped_fmu, fmu, fmu_full_path):
+    """ Function utilized by two base classes and load_fmu that does the
+        error checking for the three input arguments named 'allow_unzipped_fmu', 'fmu' and
+        the constructed variable 'fmu_full_path'.
+    """
+    if allow_unzipped_fmu:
+        if not os.path.isdir(fmu):
+            msg = "Argument named 'fmu' must be a directory if argument 'allow_unzipped_fmu' is set to True."
+            raise FMUException(msg)
+        if not os.path.isfile(os.path.join(fmu, 'modelDescription.xml')):
+            err_msg = "Specified fmu path '{}' needs".format(fmu)
+            err_msg += " to contain a modelDescription.xml according to the FMI specification."
+            raise FMUException(err_msg)
+    else:
+        # Check that the file referenced by fmu is appropriate
+        if not fmu_full_path.endswith('.fmu' if isinstance(fmu_full_path, str) else encode('.fmu')):
+            raise FMUException("FMUModel must be instantiated with an FMU (.fmu) file.")
```

Comment:
Can we replace FMUModel with some thing else? Or someone change the wording so that it better fits all classes?

Reply:
I didn't change the default error message, but perhaps the following would work? Let me know and I'll change it when we have consensus:
"Instantiating an FMU requires an FMU (.fmu) file, specified file has extension {}".format(os.path.splittext(fmu_full_path)[1])"

Reply:
Looks great!